File: Progressio_backend/Progressio.Services/Database/Entities/Character.cs
namespace Progressio.Services.Database.Entities;

public class Character
{
    public int Id { get; set; }
    public int ContentId { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    public bool IsMainCharacter { get; set; }

    public Content Content { get; set; } = null!;
    public ICollection<CharacterVote> Votes { get; set; } = [];
}

File: ui/progressio_mobile/lib/model/character.dart


class Character {
  final int id;
  final int contentId;
  final String? contentTitle;
  final String name;
  final String? description;
  final String? imageUrl;
  final bool isMainCharacter;

  Character({
    this.id = 0,
    this.contentId = 0,
    this.contentTitle,
    this.name = '',
    this.description,
    this.imageUrl,
    this.isMainCharacter = false,
  });

  factory Character.fromJson(Map<String, dynamic> json) {
    return Character(
      id: json['id'] ?? 0,
      contentId: json['contentId'] ?? 0,
      contentTitle: json['contentTitle'],
      name: json['name'] ?? '',
      description: json['description'],
      imageUrl: json['imageUrl'],
      isMainCharacter: json['isMainCharacter'] ?? false,
    );
  }
}